roles/docker/tasks/main.yml

```diff
@@ -80,13 +86,42 @@
 
 # XXX: service module doesn't do daemon-reload yet, so need to use shell module here
 # https://github.com/ansible/ansible-modules-core/issues/191
-- name: restart docker
+- name: reload docker systemd configuration
   #service: name=docker state=restarted
-  shell: sudo systemctl daemon-reload && sudo systemctl restart docker
+  shell: sudo systemctl daemon-reload
   when: "(docker_service_state.stderr | match('.*docker.service changed on disk.*')) or (docker_service_state.stdout != 'Active: active')"
   tags:
     - prebake-for-dev
 
+# XXX: this needs to happen twice after setting up the thin provisioner because
+#      of some docker bug I've not investigated.
+- name: restart docker (first time)
+  service:
+    name: docker
+    state: restarted
+  when: thin_provisioned|changed
+  ignore_errors: true
+  tags:
+    - prebake-for-dev
+
+- name: ensure docker is started
+  service:
+    name: docker
+    state: started
+  tags:
+    - prebake-for-dev
+
+- stat: path=/var/docker_images
+  register: docker_images
+  tags:
+    - prebake-for-dev
```

Comment:
we can remove the `prebake-for-dev` tag from this task as it is runtime configuration